src/worker/tasks/sendTransactionWorker.ts

```diff
@@ -181,57 +190,95 @@ const _sendUserOp = async (
 const _sendTransaction = async (
   job: Job,
   queuedTransaction: QueuedTransaction,
-): Promise<SentTransaction | ErroredTransaction> => {
+): Promise<SentTransaction | ErroredTransaction | null> => {
   assert(!queuedTransaction.isUserOp);
 
-  const { queueId, chainId, from } = queuedTransaction;
+  if (_hasExceededTimeout(queuedTransaction)) {
+    // Fail if the transaction is not sent within the specified timeout.
+    return {
+      ...queuedTransaction,
+      status: "errored",
+      errorMessage: `Exceeded ${queuedTransaction.timeoutSeconds}s timeout`,
+    };
+  }
+
+  const { queueId, chainId, from, to, overrides } = queuedTransaction;
+  const chain = await getChain(chainId);
+
   // Populate the transaction to resolve gas values.
   // This call throws if the execution would be reverted.
   // The nonce is _not_ set yet.
-  const [populatedTransaction, erroredTransaction] =
-    await getPopulatedOrErroredTransaction(queuedTransaction);
 
-  if (erroredTransaction) {
+  let populatedTransaction: PopulatedTransaction;
+  try {
+    populatedTransaction = await toSerializableTransaction({
+      from: getChecksumAddress(from),
+      transaction: {
+        client: thirdwebClient,
+        chain,
+        ...queuedTransaction,
+        to: getChecksumAddress(to),
+        // Use a dummy nonce since we override it later.
+        nonce: 1,
+
+        // Apply gas setting overrides.
+        // Do not set `maxFeePerGas` to estimate the onchain value.
+        gas: overrides?.gas,
+        maxPriorityFeePerGas: overrides?.maxPriorityFeePerGas,
+      },
+    });
+  } catch (e: unknown) {
+    const erroredTransaction: ErroredTransaction = {
+      ...queuedTransaction,
+      status: "errored",
+      errorMessage: `${e}`,
+    };
     job.log(
-      `Failed to validate transaction: ${erroredTransaction.errorMessage}`,
+      `Failed to populate transaction: ${erroredTransaction.errorMessage}`,
     );
     return erroredTransaction;
   }
 
+  // Handle if `maxFeePerGas` is overridden.
+  // Set it if the transaction will be sent, otherwise delay the job.
+  if (overrides?.maxFeePerGas && populatedTransaction.maxFeePerGas) {
+    if (overrides.maxFeePerGas > populatedTransaction.maxFeePerGas) {
+      populatedTransaction.maxFeePerGas = overrides.maxFeePerGas;
+    } else {
+      const retryAt = _minutesFromNow(5);
```

Comment:
why 5 minutes?

Reply:
I chose a reasonable period when gas values may be different enough without waiting too long. Were you thinking of something dynamic (exp backoff?) or configurable?